src/pyscipopt/scip.pxi

```diff
@@ -975,10 +975,10 @@ cdef class Constraint:
                 and self.scip_cons == (<Constraint>other).scip_cons)
 
 
-cdef void relayMessage(SCIP_MESSAGEHDLR *messagehdlr, FILE *file, const char *msg):
+cdef void relayMessage(SCIP_MESSAGEHDLR *messagehdlr, FILE *file, const char *msg) noexcept:
     sys.stdout.write(msg.decode('UTF-8'))
 
-cdef void relayErrorMessage(void *messagehdlr, FILE *file, const char *msg):
+cdef void relayErrorMessage(void *messagehdlr, FILE *file, const char *msg) noexcept:
```

Comment:
Hey @mkoeppe! Sorry for not looking into your stuff, but it flies a bit over my head... Nevertheless, there are only 3 lines where you add ```noexcept``` where there's no ```with gil```. Is that intentional?

I also find it very sad that the test coverage decreases so much, but that's really not an actual problem :)

Reply:
Okay, looking at the Cython page makes me think this is also good.

Reply:
> where you add `noexcept` where there's no `with gil`

I started by adding `noexcept`s one by one to remove the compiler errors. 
I switched to taking care of the `noexcept with gil` by global replace. 
The `noexcept` where there's no `with gil` are all needed.